Libs/Core/ctkUtils.h

```diff
@@ -30,6 +30,23 @@
 
 #include "ctkCoreExport.h"
 
+/// This macro can be used instead of Q_ASSERT to log a warning when
+/// a condition fails. It logs a warning message in debug mode and
+/// has no effect in release mode.
+#ifndef NDEBUG
+#   define CTK_VALIDATE(condition) \
```

Comment:
What about `CTK_CHECK_CONDITION` ? 

Looking at Qt Creator source code, they have "SOFT ASSERT", see https://github.com/qt-creator/qt-creator/blob/527484bba9057e3fe30c9fff2a53b556ef23b81c/src/libs/utils/qtcassert.h  and they defined `QTC_ASSERT`, we could simply have `CTK_ASSERT`

Reply:
We already use CHECK_... macros for testing (they return with EXIT_FAILURE if condition false), so I would not use the same word for a completely different purpose.

I would be OK with CTK_ASSERT name, but it may be a bit misleading. What do you think about CTK_SOFT_ASSERT?

Reply:
`CTK_SOFT_ASSERT` :+1: